feat(cli): getinfo enhancements

rsercano opened the pull request:
resolves #1059

rsercano commented:
`actions.go` line 36 there're chain & network information checks in these lines, but I didn't change anything related with chain as you can see in the changed files. 

Still, somehow it doesn't receive chain & network information.. Could you please help me with that @moshababo

rsercano commented:
This PR also includes changes from #1174. I have a few quick questions;

- Where should I get alias of xud, is there an algorithm to generate it?
- Somehow address of the local xud is empty in pool class, is there something wrong with my setup or with these changes ?
- Raiden doesn't provide chains as LND does, so I set `XUD <network>` hardcoded, is that okay ?

kilrau commented:
>     * Where should I get alias of xud, is there an algorithm to generate it?

There will be soon: https://github.com/ExchangeUnion/xud/issues/486#issuecomment-528392857

> Somehow address of the local xud is empty in pool class, is there something wrong with my setup or with these changes ?

Check `xud.conf`, if nothing configured there and you are running a lonely `xud`, it's empty I believe. The SimNet setup has to give you a TOR address.

> Raiden doesn't provide chains as LND does, so I set XUD <network> hardcoded, is that okay ?

No... Worst case you can judge by the registered `DAI` and `WETH` addresses which network it is (see ` lib/db/seeds`).

> Here's how it seems with these changes; @kilrau

Address should have a line break before the @ to make it fit in a regular CLI terminal size. Rest looks good.

rsercano commented:
Added line break just before the `@` as you said Kilian but I can't see how I can find out raiden chain ? The address printed by get info is different from registered addresses in seed folder ?

rsercano commented:
Nvm, I suppose I found it, needs a review tho.

michael1011 commented on lib/cli/commands/getinfo.ts:
```diff
@@ -3,109 +3,87 @@ import { Arguments } from 'yargs';
-    displayChainsList(info.chainsList, asset);
-  }
+  basicInfotable.push(
+    { [colors.blue('Version')]: info.version ? info.version : ''   },
```
```suggestion
    { [colors.blue('Version')]: info.version || ''   },
```
This can be simplified a little

michael1011 commented on lib/cli/commands/getinfo.ts:
```diff
@@ -3,109 +3,87 @@ import { Arguments } from 'yargs';
+  basicInfotable.push(
+    { [colors.blue('Version')]: info.version ? info.version : ''   },
+    { [colors.blue('Address')]: address },
+    { [colors.blue('Alias')] : info.alias ? info.alias : '' },
```
```suggestion
    { [colors.blue('Alias')] : info.alias || '' },
```
Same here

michael1011 commented on lib/cli/commands/getinfo.ts:
```diff
@@ -3,109 +3,87 @@ import { Arguments } from 'yargs';
+    { [colors.blue('Address')]: address },
+    { [colors.blue('Alias')] : info.alias ? info.alias : '' },
+    { [colors.blue('Channels')] :  `Active: ${info.channels ? info.channels['active'] : 0} | Pending: ${info.channels ? info.channels['pending'] : 0}\
+ | Closed: ${info.channels ? info.channels['inactive'] : 0}` },
```
Inactive channels are not `closed`. That just means that the peer is not online right now

rsercano replied:
Then how can I receive the closed ones ?

michael1011 commented on lib/cli/commands/getinfo.ts:
```diff
@@ -3,109 +3,87 @@ import { Arguments } from 'yargs';
     { [colors.blue('Address')]: info.address },
-    { [colors.blue('Channels')]: info.channels },
-    { [colors.blue('Error')]: info.error },
+    { [colors.blue('Channels')]:  `Active: ${info.channels ? info.channels : 0} | Pending: 0 | Closed: 0` },
```
Raiden has a `closed` channel state. This could be shown here.

michael1011 commented on lib/lndclient/LndClient.ts:
```diff
@@ -243,17 +244,23 @@ class LndClient extends SwapClient {
+        if (channels.active > 0) {
+          status = SwapClientStatus.Ready;
+        } else {
+          error = errors.LND_IS_UNAVAILABLE(ClientStatus.NotInitialized).message;
```
I am not sure whether we should call a client with no clients `NotInitialized`. @kilrau what do you think?

kilrau replied:
Agree, this is not the correct error message. How about  `ClientStatus.NoChannels` ?

michael1011 commented on lib/raidenclient/RaidenClient.ts:
```diff
@@ -251,20 +251,29 @@ class RaidenClient extends SwapClient {
     let error: string | undefined;
-    const version = ''; // Intentionally left blank until Raiden API exposes it
+    let status: SwapClientStatus = SwapClientStatus.Error;
+    const version = 'v0.100.3 - Rosemary'; // Intentionally left hardcoded until Raiden API exposes it
```
We have to remember to update this version regularly if we want to hardcode it.

kilrau replied:
Let's never hardcode anything unless there is no other way. I checked the Raiden Client endpoints and indeed I couldn't find one returning the version. BUT each raiden log starts by printing the version in line 0: `Welcome to Raiden, version 0.100.5a1.dev157+geb2af878d!`

rsercano replied:
I couldn't find the mentioned version description in any response of raiden API, did you get it from somewhere else @kilrau ?

michael1011 replied:
I think @kilrau means that the first row of the **log** of Raiden shows the version number. But reading that isn't a viable long term solution. I would just leave the Raiden version blank or set it to `v1` (because we are using the version 1 of their REST API)

kilrau replied:
Agree, that it is not a long term solution but would want to use it, as it's the only way we have right now. I'll open an issue on raiden side to have something similar to `getinfo` with version and so on. Once that's there, we can switch. Deal?

michael1011 replied:
@rsercano we just agreed on the call to implement a Raiden endpoint that shows the version. Because the swaps will only be working on our Raiden fork we can easily and quickly merge that new endpoint in there. I will implement it and ping you again once it is ready.

rsercano replied:
Great, thanks @michael1011 :)

michael1011 replied:
@rsercano it was easier than expected: https://github.com/ExchangeUnion/raiden/commit/2ac24318407084866bd81c0f212ae63a2cbfab33
![image](https://user-images.githubusercontent.com/8138838/64545932-fc359280-d329-11e9-907d-cedc1e982188.png)

rsercano commented:
Thanks @michael1011 and @kilrau I'll check the rest on Monday

kilrau commented on lib/raidenclient/RaidenClient.ts:
```diff
@@ -251,20 +251,29 @@ class RaidenClient extends SwapClient {
       try {
         channels = (await this.getChannels()).length;
         address = this.address;
+        if (channels > 0) {
```
Same here.

rsercano commented:
Added requested changes @kilrau @michael1011. Also I've realized that I could get closed channels from LND api via `ClosedChannels` call, so I've implemented it too.

![image](https://user-images.githubusercontent.com/3640368/64526228-08f2c000-d303-11e9-876b-ddf405a74b94.png)

rsercano commented:
Added fetching version function, it should work as soon as https://github.com/ExchangeUnion/raiden/pull/10 is merged & deployed.

sangaman commented on lib/swaps/SwapClient.ts:
```diff
@@ -10,6 +10,7 @@ enum ClientStatus {
   ConnectionVerified,
   OutOfSync,
   WaitingUnlock,
+  NoChannels,
```
I don't think this is the place for this `NoChannels` status, we're never actually setting the status of a client to `NoChannels` and if we did we'd have to alter logic in other places that depends on `ConnectionVerified` to ensure we can make calls to lnd (including calls to open channels).

rsercano replied:
Moved it to errors which makes more sense, thanks Daniel

sangaman commented on proto/xudrpc.proto:
```diff
@@ -380,20 +385,24 @@ message SwapFailure {
 message GetInfoRequest {}
 message GetInfoResponse {
-  // The version of this instance of xud.
-  string version = 1 [json_name = "version"];
```
Not a huge deal since we are still in pre-release stages, but generally speaking going forward we should be careful not to change numbers of existing fields because it would break client code that's using the old proto definition. In this case we already have `version`. It's also just reducing unnecessary code changes. Same for some of the other fields (`num_peers`, `num_pairs`, etc).

rsercano replied:
Good point

sangaman commented on proto/xudrpc.proto:
```diff
@@ -455,16 +464,19 @@ message LndChannels {
-  repeated string uris = 5 [json_name = "uris"];
-  string version = 6 [json_name = "version"];
-  string alias = 7 [json_name = "alias"];
+  SwapClientStatus status = 1 [json_name = "status"];
```
Same here for renumbering all these fields.

rsercano replied:
At first I thought we want the requested order, but you're right.

sangaman commented on proto/xudrpc.proto:
```diff
@@ -455,16 +464,19 @@ message LndChannels {
   uint32 inactive = 2 [json_name = "inactive"];
   // The number of channels that are pending on-chain confirmation before they can be used.
   uint32 pending = 3 [json_name = "pending"];
+  // The number of channels that are pending on-chain confirmation before they can be used.
```
This comment should say "The number of channels that have been closed" or something like that.

sangaman commented on proto/xudrpc.proto:
```diff
@@ -596,11 +608,19 @@ message PlaceOrderEvent {
   }
 }
 
+message RaidenChannels {
```
Can we rename `LndChannels` to `Channels` and just use that single message?

rsercano replied:
I'm commenting out setting `settled` for raiden then ?

sangaman commented on proto/xudrpc.proto:
```diff
@@ -596,11 +608,19 @@ message PlaceOrderEvent {
+}
+
 message RaidenInfo {
-  string error = 1 [json_name = "error"];
```
Same here for renumbering fields.

sangaman commented on proto/xudrpc.proto:
```diff
@@ -596,11 +608,19 @@ message PlaceOrderEvent {
-  uint32 channels = 3 [json_name = "channels"];
-  string version = 4 [json_name = "version"];
+  SwapClientStatus status = 1 [json_name = "status"];
+  string error = 2 [json_name = "error"];
```
I thought we wanted to remove `error` and replace with `status`?

rsercano commented:
Thanks for your reviews @sangaman, in fact I'm not available before Monday, due to a travel abroad, sorry about that.

rsercano commented:
Changes have been implemented @sangaman, this's the current view;
![image](https://user-images.githubusercontent.com/3640368/65241250-60531600-dae3-11e9-906d-2ee9acaca543.png)
![image](https://user-images.githubusercontent.com/3640368/65241271-6943e780-dae3-11e9-8eca-741f65ee7dc7.png)

kilrau commented:
Looks great! Questions:
* Why is `xud`'s address field empty?
* Can you run our current raiden version,which has a version endpoint? Would also like to see the fields in there.
* I suggest renaming `Chain` -> `Network` for lndbtc, lndltc, raiden too. To be consistent with xud.

rsercano commented:
> Why is xud's address field empty?

My configuration was wrong, seems okay now.
![image](https://user-images.githubusercontent.com/3640368/65322853-30217b00-dba8-11e9-9cde-cc402551eaf6.png)

> Can you run our current raiden version,which has a version endpoint? Would also like to see the fields in there.

![image](https://user-images.githubusercontent.com/3640368/65324097-7e844900-dbab-11e9-92c7-5060e91965b7.png)

> I suggest renaming Chain -> Network for lndbtc, lndltc, raiden too. To be consistent with xud.

Consider it done.

kilrau commented:
> > Why is xud's address field empty?
> 
> My configuration was wrong, seems okay now.
> ![image](https://user-images.githubusercontent.com/3640368/65322853-30217b00-dba8-11e9-9cde-cc402551eaf6.png)

Can we do the line break before the `@` here too (just ? If it's a long .onion address it'll break things again.

> > Can you run our current raiden version,which has a version endpoint? Would also like to see the fields in there.
> 
> ![image](https://user-images.githubusercontent.com/3640368/65324097-7e844900-dbab-11e9-92c7-5060e91965b7.png)

Don't really see it here, but I assume `raiden.version` calls the new version endpoint. Also, can we call it `raiden simnet` to be in line with `bitcoin simnet` and `litecoin simnet` below? 

> > I suggest renaming Chain -> Network for lndbtc, lndltc, raiden too. To be consistent with xud.
> 
> Consider it done.

:+1:

rsercano commented:
![image](https://user-images.githubusercontent.com/3640368/65325330-d6707f00-dbae-11e9-9b24-b84fb91e3a05.png)

> Don't really see it here, but I assume raiden.version calls the new version endpoint. Also, can we call it raiden simnet to be in line with bitcoin simnet and litecoin simnet below?

Yes it does. Added `raiden` hardcoded just before the network if the `tokenAddress` doesn't have a name.
![image](https://user-images.githubusercontent.com/3640368/65325920-7549ab00-dbb0-11e9-8e2f-1272656e0f1a.png)

kilrau commented:
Damn, I just found one more thing: In https://github.com/ExchangeUnion/xud/pull/1203 we finally settled on "Node Key" to describe `xud`'s public key. Can we "Public Key" -> "Node Key"? Thanks.

rsercano commented:
done @kilrau

kilrau commented:
Final ACK & merge by @sangaman
